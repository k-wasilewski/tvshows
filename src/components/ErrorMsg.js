import React from 'react';
import {useStyles} from "./useStyles";

export function ErrorMsg(props) {
    const classes = useStyles();

    return (
        <span className={classes.errorMsg}>{props.msg}</span>
    )
}