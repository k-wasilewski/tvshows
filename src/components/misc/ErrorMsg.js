import React from 'react';
import {styles} from '../../styles/styles';

export function ErrorMsg(props) {
    const classes = styles();

    return (
        <span className={classes.errorMsg}>{props.msg}</span>
    )
}