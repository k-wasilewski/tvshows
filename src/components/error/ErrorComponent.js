import React from 'react';
import {styles} from "../../styles/styles";
import {useHistory} from 'react-router-dom';
import {ErrorMsg} from './ErrorMsg';
import Button from '@material-ui/core/Button';

export function ErrorComponent(props) {
    const history = useHistory();
    const classes = styles();

    return (
        <div className={classes.dashboard}>
            <ErrorMsg msg={props.msg} />
            <Button variant='outlined' size='small' onClick={() => history.push('/')}>
                Powr√≥t</Button>
        </div>
    );
};

export default ErrorComponent;
