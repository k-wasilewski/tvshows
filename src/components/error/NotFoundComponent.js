import React from 'react';
import {styles} from "../../styles/styles";
import {useHistory} from 'react-router-dom';
import {ErrorMsg} from './/ErrorMsg';
import Button from '@material-ui/core/Button';

export function NotFoundComponent(props) {
    const history = useHistory();
    const classes = styles();

    return (
        <div className={classes.dashboard}>
            <ErrorMsg msg={'Nie znaleziono strony'} />
            <Button variant='outlined' size='small' onClick={() => history.push('/')}>
                Powr√≥t</Button>
        </div>
    );
};

export default NotFoundComponent;
