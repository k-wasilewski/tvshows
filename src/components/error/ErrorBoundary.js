import React from "react";
import ErrorComponent from "./ErrorComponent";

export default class ErrorBoundary extends React.Component {
    state = {
        errorMessage: ''
    };

    static getDerivedStateFromError(error) {
        return {errorMessage: error.toString()};
    }

    componentDidCatch(error, info) {
        this.logErrorToServices(error.toString(), info.componentStack);
    }

    logErrorToServices = console.log;   //better logging

    render() {
        if (this.state.errorMessage) {
            return (
                <ErrorComponent msg='Wystąpił błąd w aplikacji' />
            );
        }
        return this.props.children;
    };
};