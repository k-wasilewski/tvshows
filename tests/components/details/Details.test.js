import React from 'react';
import renderer from 'react-test-renderer';
import store from "../../../src/redux/store";
import {BrowserRouter} from "react-router-dom";
import ConnectedDetails, {Details} from "../../../src/components/details/Details";
import {Provider} from "react-redux";
import {configure, mount} from 'enzyme';
import Adapter from "enzyme-adapter-react-16";
import {ErrorMsg} from "../../../src/components/error/ErrorMsg";
import Button from '@material-ui/core/Button';

describe("Details rendering specification", () => {
    it('Details is rendered', () => {
        const testResult = {
            show: {
                name: 'sample name',
                image: {
                    medium: 'mediumSrc',
                    original: 'originalSrc'
                }
            },
            score: 2
        };

        const component = renderer.create(
            <Provider store={store}>
                <BrowserRouter>
                    <ConnectedDetails detailedResult={testResult} />
                </BrowserRouter>
            </Provider>
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

const mockHistoryPush = jest.fn();

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useHistory: () => ({
        push: mockHistoryPush,
    }),
}));

describe("Details functional specification", () => {
    let component;

    beforeEach(() => {
        configure({ adapter: new Adapter() });
    });

    afterEach(() => {
        component.unmount();
    });

    it('renders ErrorMsg when props.detailedResult value is empty', (done) => {
        const testResult = {
            show: {
                name: 'sample name',
                image: {
                    medium: 'mediumSrc',
                    original: 'originalSrc'
                }
            },
            score: 2
        };

        const errorMsg = 'Należy wybrać element z listy';

        component = mount(
            <Provider store={store}>
                <BrowserRouter>
                    <ConnectedDetails />
                </BrowserRouter>
            </Provider>
        );

        expect(component.find(ErrorMsg).html()).toContain(errorMsg);
        component.setProps({children: <BrowserRouter>
                <Details detailedResult={testResult}/>
            </BrowserRouter>});
        component.update();

        setTimeout(function () {
            expect(component.find(ErrorMsg).debug()).toBe('');
            done();
        }, 500);
    });

    it('redirects to \' when goBackBtn is clicked', (done) => {
        component = mount(
            <Provider store={store}>
                <BrowserRouter>
                    <ConnectedDetails />
                </BrowserRouter>
            </Provider>
        );

        const goBackBtn = component.find(Button).at(0);
        goBackBtn.simulate('click');
        component.update();

        setTimeout(function () {
            expect(mockHistoryPush).toHaveBeenCalledWith('/');
            done();
        }, 500);
    });
});