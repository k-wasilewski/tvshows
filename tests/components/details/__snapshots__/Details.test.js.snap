// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details rendering specification Details is rendered 1`] = `
<div
  className="makeStyles-dashboard-10"
>
  <span
    className="makeStyles-errorMsg-6"
  >
    Należy wybrać element z listy
  </span>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSizeSmall MuiButton-sizeSmall"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Powrót
    </span>
  </button>
</div>
`;
